repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkIMain.scala

```diff
@@ -978,15 +978,32 @@ import org.apache.spark.annotation.DeveloperApi
   @DeveloperApi
   def reset() {
     clearExecutionWrapper()
-    resetClassLoader()
     resetAllCreators()
+    resetToCheckpoint()
+  }
+  /**
+   * Save  previous Requests,referenced Names  and defined Names
+   * later these values can be used in reset operation
+   */
+    @DeveloperApi
+  def saveCheckpoint() {
+    prevRequestsCheckpoint      ++= prevRequests.clone()
+    referencedNameMapCheckpoint ++= referencedNameMap.clone()
+    definedNameMapCheckpoint    ++=definedNameMap.clone()
+  }
+  /**
+   * Reset prevRequests,referencedNameMap and definedNameMap to
+   * checkpointed state, so that it will only retain  spark initialization details
+   */
+  def resetToCheckpoint() {
     prevRequests.clear()
     referencedNameMap.clear()
     definedNameMap.clear()
-    virtualDirectory.delete()
-    virtualDirectory.create()
+    // Restore to checkpointed state
```

Comment:
virtualDirectory.delete&create is to clear the already compiled classes, but this will clear sc and sql context related classes too.
If replay command need not be supported in spark-shell, can we disable it?